felix/dataplane/linux/ipip_mgr.go

```diff
@@ -86,206 +70,123 @@ type ipipManager struct {
-		localIPAMBlocks:   map[string]*proto.RouteUpdate{},
-		ipipDevice:        deviceName,
-		ipVersion:         ipVersion,
-		externalNodeCIDRs: dpConfig.ExternalNodesCidrs,
-		routesDirty:       true,
-		ipSetDirty:        true,
-		dpConfig:          dpConfig,
-		nlHandle:          nlHandle,
-		routeProtocol:     calculateRouteProtocol(dpConfig),
-		logCtx:            logrus.WithField("ipVersion", ipVersion),
-		opRecorder:        opRecorder,
-	}
-}
-
-func (m *ipipManager) OnUpdate(msg interface{}) {
-	switch msg := msg.(type) {
-	case *proto.RouteUpdate:
-		cidr, err := ip.CIDRFromString(msg.Dst)
-		if err != nil {
-			m.logCtx.WithError(err).WithField("msg", msg).
-				Warning("Unable to parse route update destination. Skipping update.")
-			return
-		}
-		if m.ipVersion != cidr.Version() {
-			// Skip since the update is for a mismatched IP version
-			return
-		}
-
-		// In case the route changes type to one we no longer care about...
-		m.deleteRoute(msg.Dst)
-
-		// Process remote IPAM blocks.
-		if isType(msg, proto.RouteType_REMOTE_WORKLOAD) && msg.IpPoolType == proto.IPPoolType_IPIP {
-			m.logCtx.WithField("msg", msg).Debug("IPIP data plane received route update")
-			m.routesByDest[msg.Dst] = msg
-			m.routesDirty = true
-		}
-
-		// Process IPAM blocks that aren't associated to a single or /32 local workload
-		if routeIsLocalBlock(msg, proto.IPPoolType_IPIP) {
-			m.logCtx.WithField("msg", msg).Debug("IPIP data plane received route update for IPAM block")
-			m.localIPAMBlocks[msg.Dst] = msg
-			m.routesDirty = true
-		} else if _, ok := m.localIPAMBlocks[msg.Dst]; ok {
-			m.logCtx.WithField("msg", msg).Debug("IPIP data plane IPAM block changed to something else")
-			delete(m.localIPAMBlocks, msg.Dst)
-			m.routesDirty = true
-		}
-	case *proto.RouteRemove:
-		// Check to make sure that we are dealing with messages of the correct IP version.
-		cidr, err := ip.CIDRFromString(msg.Dst)
-		if err != nil {
-			m.logCtx.WithError(err).WithField("msg", msg).
-				Warning("Unable to parse route removal destination. Skipping update.")
-			return
-		}
-		if m.ipVersion != cidr.Version() {
-			// Skip since the update is for a mismatched IP version
-			return
-		}
-		m.deleteRoute(msg.Dst)
+		activeHostnameToIP: map[string]string{},
+		hostname:           dpConfig.Hostname,
+		tunnelDevice:       tunnelDevice,
+		tunnelDeviceMTU:    mtu,
+		ipVersion:          ipVersion,
+		externalNodeCIDRs:  dpConfig.ExternalNodesCidrs,
+		ipSetDirty:         true,
+		dpConfig:           dpConfig,
+		routeProtocol:      calculateRouteProtocol(dpConfig),
+		logCtx: logrus.WithFields(logrus.Fields{
+			"ipVersion":    ipVersion,
+			"tunnelDevice": tunnelDevice,
+		}),
+		opRecorder: opRecorder,
+		routeMgr: newRouteManager(
+			mainRouteTable,
+			proto.IPPoolType_IPIP,
+			tunnelDevice,
+			ipVersion,
+			mtu,
+			dpConfig,
+			opRecorder,
+			nlHandle,
+		),
+	}
+
+	m.routeMgr.routeClassTunnel = routetable.RouteClassIPIPTunnel
```

Comment:
I think I'm still not sure why these aren't arguments to `newRouteManager` - presumably these are required fields, right?

Reply:
I'll address this in the next PR.